components/collator/README.md

```diff
@@ -9,113 +9,194 @@ and as part of the [`icu`](https://docs.rs/icu/latest/icu/) crate. See the latte
+assert_eq!(collator_l3.compare("Ao", "aò"), Ordering::Less);
+assert_eq!(collator_l3.compare("A", "Ⓐ"), Ordering::Less);
+```
 
-    One possible alternative to loading a different table for "search" would be
-    performing the mapping of archaic jamo to the modern approximations as a
-    special preprocessing step for the incoming characters, which would allow
-    the lookup of the resulting modern jamo from the normal root jamo table.
-
-    "searchjl" is even more problematic than "search", since "searchjl" uses
-    prefixes matches with jamo, and currently Hangul is assumed not to participate
-    in prefix or contraction matching.
+### Alternate Handling
+
+Allows alternate handling for certain customized collation orderings, including the option to
+ignore the special handling for the strings of such customizations.
+
+### Case Level
+
+Whether to distinguish case in sorting, even for sorting levels higher than tertiary,
+without having to use tertiary level just to enable case level differences.
+
+```rust
+use core::cmp::Ordering;
+use icu_collator::*;
+
+// Primary
+
+let data_provider = icu_testdata::get_provider();
+let mut options = CollatorOptions::new();
+options.set_strength(Some(Strength::Primary));
+options.set_case_level(Some(false));
+let primary =
+  Collator::try_new_with_buffer_provider(&data_provider,
+                    &Default::default(),
+                    options).unwrap();
+
+assert_eq!(primary.compare("DEAL", "ⓓⓔⓐⓛ"), Ordering::Equal);
+assert_eq!(primary.compare("ⓓⓔⓐⓛ", "déjavu"), Ordering::Less);
+assert_eq!(primary.compare("déjavu", "dent"), Ordering::Less);
+assert_eq!(primary.compare("dent", "develop"), Ordering::Less);
+
+// Primary with case level on
+
+options.set_strength(Some(Strength::Primary));
+options.set_case_level(Some(true));
+let primary_and_case =
+  Collator::try_new_with_buffer_provider(&data_provider,
+                    &Default::default(),
+                    options).unwrap();
+
+assert_eq!(primary_and_case.compare("DEAL", "ⓓⓔⓐⓛ"), Ordering::Equal);
+assert_eq!(primary_and_case.compare("ⓓⓔⓐⓛ", "déjavu"), Ordering::Less);
+assert_eq!(primary_and_case.compare("déjavu", "dent"), Ordering::Less);
+assert_eq!(primary_and_case.compare("dent", "develop"), Ordering::Less);
+
+// Tertiary
+
+options.set_strength(Some(Strength::Tertiary));
+options.set_case_level(Some(false));
+let tertiary =
+  Collator::try_new_with_buffer_provider(&data_provider,
+                    &Default::default(),
+                    options).unwrap();
+
+assert_eq!(tertiary.compare("ⓓⓔⓐⓛ", "DEAL"), Ordering::Less);
+assert_eq!(tertiary.compare("DEAL", "déjavu"), Ordering::Less);
+assert_eq!(tertiary.compare("déjavu", "dent"), Ordering::Less);
+assert_eq!(tertiary.compare("dent", "develop"), Ordering::Less);
+
+
+```
+
+### Case First
+
+Whether to swap the ordering of uppercase and lowercase.
+
+### Backward second level
+
+Compare the second level in backward order.
```

Comment:
Please add a remark here about Canadian French, since otherwise it looks a bit odd why an option like this is offered.

Reply:
Done.